File: projects/xobusy/bojs/sort.py
from requests import get
from os import chdir, getcwd, listdir, mkdir
from os.path import splitext, isfile, isdir
from shutil import move
from pathlib import Path

chdir('problems')

def move_path(frompath: str, dest: str):
  parent = Path(dest).parent.absolute()
  if isdir(parent) == False:
    mkdir(parent)
  move(frompath, dest)

for i in [i for i in listdir(getcwd()) if isfile(i)]:
  ver = get(f'https://solved.ac/api/v3/problem/lookup?problemIds={splitext(i)[0]}').json()[0]["level"]
  print(ver)
  if ver == 1:
    move_path(i, f"BronzeV/{i}")
  elif ver == 2:
    move_path(i, f"BronzeIV/{i}")
  elif ver == 3:
    move_path(i, f"BronzeIII/{i}")
  elif ver == 4:
    move_path(i, f"BronzeII/{i}")
  elif ver == 8:
    move_path(i, f"SilverIII/{i}")

File: projects/xobusy/bojs/sort.jl
using HTTP
using Base.Filesystem
import JSON.parse

if !Filesystem.isdir("problems")
  Filesystem.mkdir("problems")
end

function move_path(frompath::String, dest::String)
  p = splitdir(frompath)[1]
  if !isdir(p)
    mkdir(p)
  end
  Filesystem.mv(frompath, dest)
end

probs = Vector{String}() # uint16: 0 ~ 65535
versions = Dict(
  1=>"BronzeV",
  2=>"BronzeIV",
  3=>"BronzeIII",
  4=>"BronzeII",
  5=>"BronzeI",
  8=>"SilverIII",
  12=>"GoldIV"
)

for i in readdir("problems")
  if isfile("problems/$i")
    push!(probs, i)
  end
end

asyncmap(probs) do i
  noextension = Filesystem.splitext(i)[1]
  ver = parse(String(HTTP.get("https://solved.ac/api/v3/problem/lookup?problemIds=$noextension").body))[1]["level"]
  println(ver)
  try
    version = versions[ver]
    move_path("problems/$i", "problems/$version/$i")
  catch KeyError
    move_path("problems/$i", "problems/idks/$i")
    println("some error here")
  end
end
